kornia/geometry/bbox_v2.py

```diff
@@ -0,0 +1,602 @@
+            corners (x1, y1, z1) and (x2, y2, z2) is computed as ``width = x2 - x1``, ``height = y2 - y1`` and
+            ``depth = z2 - z1``.
+
+    Returns:
+        - Bounding box depths, shape of :math:`(N,)` or :math:`(B,)`.
+        - Bounding box heights, shape of :math:`(N,)` or :math:`(B,)`.
+        - Bounding box widths, shape of :math:`(N,)` or :math:`(B,)`.
+
+    Example:
+        >>> boxes = torch.tensor([[[ 0,  1,  2],
+        ...         [10,  1,  2],
+        ...         [10, 21,  2],
+        ...         [ 0, 21,  2],
+        ...         [ 0,  1, 32],
+        ...         [10,  1, 32],
+        ...         [10, 21, 32],
+        ...         [ 0, 21, 32]],
+        ...        [[ 3,  4,  5],
+        ...         [43,  4,  5],
+        ...         [43, 54,  5],
+        ...         [ 3, 54,  5],
+        ...         [ 3,  4, 65],
+        ...         [43,  4, 65],
+        ...         [43, 54, 65],
+        ...         [ 3, 54, 65]]]) # 2x8x3
+        >>> infer_bbox3d_shape(boxes)
+        (tensor([30, 60]), tensor([20, 50]), tensor([10, 40]))
+    """
+    _check_bbox3d_dimensions(boxes)
+
+    left = boxes[..., [1, 2, 5, 6], 0]
+    right = boxes[..., [0, 3, 4, 7], 0]
+    widths = left - right
+
+    bottom = boxes[..., [2, 3, 6, 7], 1]
+    upper = boxes[..., [0, 1, 4, 5], 1]
+    heights = bottom - upper
+
+    depths = boxes[..., 4:, 2] - boxes[..., :4, 2]
+    return depths[..., 0], heights[..., 0], widths[..., 0]
+
+
+def bbox_to_mask(boxes: torch.Tensor, width: int, height: int) -> torch.Tensor:
+    """Convert 2D bounding boxes to masks. Covered area is 1. and the remaining is 0.
+
+    Args:
+        boxes: a tensor containing the coordinates of the bounding boxes to be extracted. The tensor must have the shape
+            of Nx4x2 or BxNx4x2, where each box is defined in the following ``clockwise`` order: top-left, top-right,
+            bottom-right and bottom-left. The coordinates must be in the x, y order. The height and width of a box with
+            corners (x1, y1) and (x2, y2) is computed as ``width = x2 - x1`` and ``height = y2 - y1``.
+        width: width of the masked image/images.
+        height: height of the masked image/images.
+
+    Returns:
+        the output mask tensor, shape of :math:`(N, width, height)` or :math:`(B,N, width, height)` and dtype of boxes.
+
+    Note:
+        It is currently non-differentiable.
+
+    Examples:
+        >>> boxes = torch.tensor([[
+        ...        [1., 1.],
+        ...        [4., 1.],
+        ...        [4., 3.],
+        ...        [1., 3.],
+        ...   ]])  # 1x4x2
+        >>> bbox_to_mask(boxes, 5, 5)
+        tensor([[[0., 0., 0., 0., 0.],
+                 [0., 1., 1., 1., 0.],
+                 [0., 1., 1., 1., 0.],
+                 [0., 0., 0., 0., 0.],
+                 [0., 0., 0., 0., 0.]]])
+    """
+    _check_bbox_dimensions(boxes)
+    mask = torch.zeros((*boxes.shape[:-2], height, width), dtype=boxes.dtype, device=boxes.device)
```

Comment:
I don't know a clean approach to support scripting and without creating a lot of boilderplate code for this case `*boxes.shape[:-2]` . There are several lines similar to this. ¿How could I refactor to be torchscript friendly?

Reply:
Just ``` boxes.shape[0], boxes.shape[1]``` will do. It is 2D only right?

Reply:
It could be (B, N, 4, 2) or (N, 4, 2). A workaround would be to convert  (N, 4, 2) to (1, N, 4, 2), make all calculations and then move back to (N, 4, 2). If there isn't another option, I would do it.